src/type-utils.ts

```diff
@@ -1,19 +1,17 @@
-import * as Promise from 'bluebird';
-
 const equality = (from: string, to: string) => ['Equals', from, to];
-const checkRequired = <T>(validateFn: (value: any) => T) =>
-	Promise.method((value: any, required: boolean): typeof required extends true
-		? T
-		: T | null => {
-		if (value == null) {
-			if (required) {
-				throw new Error('cannot be null');
-			} else {
-				return null;
-			}
+const checkRequired = <T>(validateFn: (value: any) => T) => async (
+	value: any,
+	required: boolean,
+): Promise<typeof required extends true ? T : T | null> => {
```

Comment:
TS is 🤯 sometimes.

Reply:
I've rewritten it and improved the typings as well